### ISSUE TITLE
Question about the weighting factor of the entropy loss

### ISSUE THREAD

**Issue by HanqingXu** [valeoai/ADVENT#5]
Hi, 
I checked the code ,and suprised to find that the weighting factor of the entropy loss and adversarial loss are both really small(1e-3 compared with the segmentation weighting factor 1.0), so I wonder if it really works out in practice. And for the direct entropy loss, could it be possible that the network eventually predicts every pixel as one class, in which case the network would have a really low entropy loss but bad performance.
Looking foward to hearing your reply.

**Comment by tuanhungvu** [comment]
Hello @HanqingXu ,
Sorry for the late response. In some sense, the two aforementioned losses play as regularization terms to bring "adaptation effect" to the final model. Their weighting factors are indeed sensitive to the particular adaptation setting. Normally we cross-validate to choose the proper weighting values. Regarding the direct entropy loss concern, it is possible that the model gets "corrupted" as you described, especially when using too big entropy weight. Pre-training only on the source domain might mitigate such adverse effects. However, as we use unsupervised losses on the target samples, there is no good guarantee to avoid the corrupted scenario. Again, there is no magic hyper-parameters that work for all cases, you will need to cross-validate to find the good set.
T-H

**Comment by HanqingXu** [comment]
Thank your for the reply. I tried the entropy loss term in my dataset. For the direct loss, even if I tried tuning the weighting factor, it still hurts the performance. As for the adversarial loss, it doesn't even converge. I saw several methods from your paper and try weighting the unsupervised loss, and

**Comment by HanqingXu** [comment]
Thank your for the reply. I tried the entropy loss term in my dataset. For the direct loss, even if I tried tuning the weighting factor, it still hurts the performance. As for the adversarial loss, it doesn't even converge. I saw several methods from your paper and try weighting the unsupervised loss, and training on specific ranges. Neither worked. Any suggestios would be appreciated.

**Comment by tuanhungvu** [comment]
Can you specify which dataset and which task you are working on?

**Comment by HanqingXu** [comment]
I tried the KITTI dataset which has only 200 images (which was my intention since I want it to fully exploit the advantage of the domain adaptation), and my own private dataset which is lane marking dataset(8 classes) in some parking lot. It has only around 200  as well

**Comment by tuanhungvu** [comment]
Ok I see. It's actually difficult to give you suggestions on a set-up that we've never seen. Maybe some guidelines for you: - make sure that your baseline performance on the target set (no UDA) is "OKish", - pretrain the model without UDA only on source. - do more data augmentation as your datasets are quite small.
T-H
